src/webgpu/api/operation/sampling/anisotropy.spec.ts

```diff
@@ -6,14 +6,506 @@ Note that anisotropic filtering is never guaranteed to occur, but we might be ab
+        { binding: 2, resource: { buffer: this.uniformBuffer } },
+      ],
+      layout: this.pipeline.getBindGroupLayout(0),
+    });
+
+    const colorAttachment = this.device.createTexture({
+      format: kColorAttachmentFormat,
+      size: { width: kRTSize, height: kRTSize, depth: 1 },
+      usage: GPUTextureUsage.COPY_SRC | GPUTextureUsage.OUTPUT_ATTACHMENT,
+    });
+    const colorAttachmentView = colorAttachment.createView();
+
+    const encoder = this.device.createCommandEncoder();
+    const pass = encoder.beginRenderPass({
+      colorAttachments: [
+        {
+          attachment: colorAttachmentView,
+          storeOp: 'store',
+          loadValue: { r: 0.0, g: 0.0, b: 0.0, a: 1.0 },
+        },
+      ],
+    });
+    pass.setPipeline(this.pipeline);
+    pass.setVertexBuffer(0, this.vertexBuffer);
+    pass.setBindGroup(0, bindGroup);
+    pass.draw(6);
+    pass.endPass();
+    this.device.defaultQueue.submit([encoder.finish()]);
+
+    return colorAttachment;
+  }
+}
+
+export const g = makeTestGroup(SamplerAnisotropicFilteringSlantedPlaneTest);
+
+g.test('anisotropic_filter_checkerboard')
+  .desc(
+    `anisotropic filter rendering tests that draws a slanted plane and samples from a texture
+    that only has a top level mipmap, the content of which is like a checkerboard.
+    We will check the rendering result using sampler with maxAnisotropy values to be
+    different from each other, as the sampling rate is different.
+    We will also check if those large maxAnisotropy values are clamped so that rendering is the
+    same as the supported upper limit say 16.
+    A smiliar webgl demo is at https://jsfiddle.net/yqnbez24`
+  )
+  // .unimplemented();
+  // .params([
+  //   ...poptions('maxAnisotropy', [1, 2, 4, 16]),
+  // ])
+  .fn(async t => {
+    t.setup();
+
+    // init texture with only a top level mipmap
+    const textureSize = 32;
+    const texture = t.device.createTexture({
+      mipLevelCount: 1,
+      size: { width: textureSize, height: textureSize, depth: 1 },
+      format: kTextureFormat,
+      usage: GPUTextureUsage.COPY_DST | GPUTextureUsage.SAMPLED,
+    });
+
+    const textureEncoder = t.device.createCommandEncoder();
+
+    const bufferSize = kBytesPerRow * textureSize; // RGBA8 for each pixel (256 > 16 * 4)
+
+    // init checkerboard texture data
+    const data: Uint8Array = new Uint8Array(bufferSize);
+    for (let r = 0; r < textureSize; r++) {
+      const o = r * kBytesPerRow;
+      for (let c = o, end = o + textureSize * 4; c < end; c += 4) {
+        const cid = (r + (c - o) / 4) % 2;
+        const color = checkerColors[cid];
+        data[c] = color[0];
+        data[c + 1] = color[1];
+        data[c + 2] = color[2];
+        data[c + 3] = color[3];
+      }
+    }
+    const buffer = t.device.createBuffer({
+      usage: GPUBufferUsage.COPY_SRC | GPUBufferUsage.COPY_DST,
+      size: data.byteLength,
+    });
+    t.device.defaultQueue.writeBuffer(buffer, 0, data);
```

Comment:
You can use GPUTest.makeBufferWithContents to make this slightly simpler.

Unfortunately we don't have a makeTextureWithContents but it's something I'd like to do eventually.